Re #74: Security error at zorbuth

smitgd opened the pull request:
This adds support for a location string that overrides the WOEID
entry. So now can enter a free form location such as "Pine Crest
TN" or "London, england". Only when location is left blank is the
entered WOEID used in the yahoo query. Otherwise, the location
string (that can also be a zipcode) is used in the query.
Also, added note that a web search can be done to find alternate
sites to obtain WOEID if needed, plus some clean up on configuration
labels (periods at end of sentences, etc.).
    modified:   plasmoid/contents/config/main.xml
    modified:   plasmoid/contents/ui/ConfigGeneral.qml
    modified:   plasmoid/contents/ui/Yahoo.qml

smitgd committed:
Re #74: Security error at zorbuth
This adds support for a location string that overrides the WOEID
entry. So now can enter a free form location such as "Pine Crest
TN" or "London, england". Only when location is left blank is the
entered WOEID used in the yahoo query. Otherwise, the location
string (that can also be a zipcode) is used in the query.
Also, added note that a web search can be done to find alternate
sites to obtain WOEID if needed, plus some clean up on configuration
labels (periods at end of sentences, etc.).
	modified:   plasmoid/contents/config/main.xml
	modified:   plasmoid/contents/ui/ConfigGeneral.qml
	modified:   plasmoid/contents/ui/Yahoo.qml

librehat commented:
I'd prefer if we can have a radio button to choose between WOEID and location. In this way, we only need one text entry in configuration tab. If user changes from WOEID to location (or vice versa), then the entry should clear its text.

smitgd commented:
Yes, that does seem better.  Will make the change.

smitgd committed:
As requested, make selection between entry of location vs. woeid
via radio buttons (two buttons). This allows just one text entry
field for location or woeid depending on radio button selection.
Implemented, also as requested, that selection of the currently
unselected radio button shall blank the loc/woeid text field.
Also added an additional feature so that the previous saved
content of the blanked field is restored if the original radio
button is re-selected.
	modified:   plasmoid/contents/config/main.xml
	modified:   plasmoid/contents/ui/ConfigGeneral.qml
	modified:   plasmoid/contents/ui/Yahoo.qml

smitgd commented:
Added the radio buttons to select location vs. woeid  and reduced from 2 to 1 the number of text entry fields and it now does double duty as woeid or location depending on radio button selection.  Also implemented the blanking feature you requested. However, I also added a feature to restore the blanked content if the original radio button is re-selected. This avoids having the previously saved entry disappear when the other radio button is selected, possibly by accident. 
Since the previous commit "worked", I did not rebase/squash it out of the history. So this pull req. now contains 2 commits.

smitgd committed:
Cause TextField id woeidField label to toggle between "WOEID" and "Location"
instead of having static label "Location or WOEID" so that it follows
the radio button selection.
Sorry, also left off "onLinkActivated" action for label containing zourbuth
link.
	modified:   plasmoid/contents/ui/ConfigGeneral.qml

smitgd commented:
Added one more small feature to this PReq. This causes the label for the woeid/location text field to toggle between WOEID and Location depending on the current radio button selection. Before the label was just the static string "Location or WOEID".  Seems to work OK.
After adding this, noticed that the link to the zourbuth site wasn't working. I had accidentially removed the onLinkActivated action. So now put it back and link now works again.

librehat commented:
Sorry it has been very busy for me recently, I'll try to spare some time to review your code asap

smitgd commented:
No problem.

librehat commented on plasmoid/contents/ui/ConfigGeneral.qml:
```diff
@@ -17,22 +17,76 @@ Item {
+        if (woeChecked && plasmoid.configuration.locationEntry) { 
+            cfg_woeid = plasmoid.configuration.woeid
+        } else if (!locChecked) {
+                cfg_woeid = ""
```
Please fix the indentation here. same for L43

smitgd replied:
OK. But, as recommended, no longer assigning to cfg_woeid.

librehat commented on plasmoid/contents/ui/ConfigGeneral.qml:
```diff
@@ -17,22 +17,76 @@ Item {
         RowLayout {
             Label {
-                text: i18n("WOEID")
+                text: woeChecked ? i18n("WOEID") : i18n("Location")
```
I don't think we need to show label here since the radio button already tells user what it's being entered

smitgd replied:
This causes the label to match the radio button selection. I think it helps the user know whether to enter a location or woeid in the text box below the radio buttons. But if you really object to this I will remove it.
I have added comment explaining what this does.

librehat commented on plasmoid/contents/ui/ConfigGeneral.qml:
```diff
@@ -17,22 +17,76 @@ Item {
+    property bool woeChecked: plasmoid.configuration.woeidEntry
+
+    function onLocClicked() {
+        if (woeChecked && plasmoid.configuration.locationEntry) { 
```
The if-else block here is confusing, could you make the logic here clearer?

smitgd replied:
I have added comment explaining what this does.

librehat commented on plasmoid/contents/ui/ConfigGeneral.qml:
```diff
@@ -17,22 +17,76 @@ Item {
     property alias cfg_interval: intervalField.text
     property alias cfg_timeFormat24: timeFormat24Field.checked
     property alias cfg_useWxFonts: useWxFontsField.checked
+    property bool locChecked: plasmoid.configuration.locationEntry
```
Why do you need these? `cfg_locationEntry` is automatically bound.

smitgd replied:
locChecked and woeChecked keep the state of the radio buttons for use in the "on clicked" functions. They seem to be needed since once you are inside the on-clicked functions the plasmoid.configuraation.locationEntry/woeidEntry values match the button state clicked on. These addiitional variables hold the "current" state right before the button is clicked.
So I think this are needed. (Actually, one variable would be enough since these are always the binary inverse of each other. However, for clearer logic, I think having two variables is better.)

librehat commented on plasmoid/contents/ui/ConfigGeneral.qml:
```diff
@@ -17,22 +17,76 @@ Item {
+
+    function onLocClicked() {
+        if (woeChecked && plasmoid.configuration.locationEntry) { 
+            cfg_woeid = plasmoid.configuration.woeid
```
You don't assign values to `cfg_*` manually. `cfg_` is a special prefix for Plasma and the configuration value will be set automatically (since we have already declared it as a `alias`).

smitgd replied:
OK. To cause the text field to clear or be restored I am now assigning to woeidField.text instead of cfg_woeid.

smitgd committed:
Clean-ups requested by Symeon:
1. Add comments to onLocClicked() and onWoeClicked()
2. No longer assign value to alias var cfg_woeid (in on-click functions).
	modified:   plasmoid/contents/ui/ConfigGeneral.qml

smitgd commented:
Added another commit to this PReq which I hope covers your concerns.
-gene

librehat commented:
I personally feel the logic there is a bit redundant. If we just show an entry, then we don't need to process onClicked event. Instead, we just need to read which radio button is selected, which means no strange bindings and value assignments.

smitgd commented:
Have you tried running it with my latest commit and, if so, is the functionality OK with you?
Please notice that when the original unselected button is selected the text field clears. But if the original button is re-selected, before Apply or OK clicked, (even if new text is entered) the original text content is restored (the last saved woeid or location). 
Before I attempt any more optimizations or changes I want to make sure that this functionality is OK.
Thanks,
-gene
